plasmapy/analysis/swept_langmuir/ion_saturation_current.py

```diff
@@ -0,0 +1,240 @@
+
+    current_bound: `float`
+        A fraction representing a percentile window around the minimum
+        current.  The points to be fitted are defined to be within this
+        window.  For example, a value of ``0.1`` indicates to use all
+        points within 10% of the minimum current.  (DEFAULT ``None``)
+
+        |
+
+        If neither ``current_bound`` or ``voltage_bound`` are specified,
+        then the routine will collect indices based on an internal
+        ``current_bound`` setting for the specified ``fit_type``.
+
+        +-----------------------+--------------------------------------+
+        | ``"linear"``          | 0.4                                  |
+        +-----------------------+--------------------------------------+
+        | ``"exponential"``     | 1.0                                  |
+        +-----------------------+--------------------------------------+
+        | ``"exp_plus_linear"`` | 1.0                                  |
+        +-----------------------+--------------------------------------+
+
+        |
+
+        Cannot be used with keyword ``voltage_bound``.
+
+    voltage_bound: `float`
+        A bias voltage (in volts) that specifies an upper bound used to
+        collect the points for the curve fit.  That is, points that
+        satisfy ``voltage <= voltage_bound`` are used in the fit.
+        (DEFAULT ``None``)
+
+        |
+
+        Cannot be used with keyword ``current_bound``.
+
+    Returns
+    -------
+    isat: `~plasmapy.analysis.fit_functions.Linear`
+        A :term:`fit-function` representing the linear portion of the
+        fitted curve.
+
+    extras: `ISatExtras`
+        Additional information from the curve fit:
+
+        * ``extras.fitted_func`` is the :term:`fit-function` (specified
+          by ``fit_type``) fitted to the IV-curve
+        * ``extras.rsq`` is the coefficient of determination
+          (r-squared) value of the ``extras.fitted_func`` to the IV-curve
+        * ``extras.fitted_indices`` is a `slice` object representing the
+          points used in the curve fit (i.e.
+          ``(voltage[extras.fitted_indices], current[extras.fitted_indices])``).
+
+    Notes
+    -----
+    This routine works by:
+
+    1. Selecting the points to be used in the fit as determined by
+       ``voltage_bound`` or ``current_bound``.
+    2. Fitting the selected points with the :term:`fit-function`
+       specified by ``fit_type``.
+    3. Extracting the linear component of the fit and returning that as
+       the ion-saturation current.
+
+    This routine opts to return a function representing a linear
+    ion-saturation current, since, while ideal planar Langmuir probes
+    reach a steady-state ion-saturation current, real world Langmuir
+    probes "suffer" from expanding sheaths as the bias voltage
+    becomes increasingly negative.  This sheath expansion results the
```

Comment:
```suggestion
    becomes increasingly negative.  This sheath expansion results in the
```
This my only quasi-required comment, so if you fix this I can approve it.  All of my other suggestions are either optional or hilarious.